app/models/user.rb

```diff
@@ -15,6 +15,36 @@ class User < ActiveRecord::Base
 
   scope :free_users, -> (project) { User.where.not(id: (project.user_ids + [project.owner.id])).collect {|p| [ p.name, p.id ]  } }
 
+
+  def admin_permissions?
+    admin
+  end
+
+  def organization_admin_permissions?(organization_id)
+    admin_permissions? || OrganizationsUser.where(user_id: self.id, organization_id: organization_id).first.try(:admin?)
+  end
+
+  def organization_member_permissions?(organization_id)
+    admin_permissions? || OrganizationsUser.where(user_id: self.id, organization_id: organization_id).first.present?
+  end
+
+  def project_admin_permissions?(project_id, organization_id)
+    organization_admin_permissions?(organization_id) || (organization_member_permissions?(organization_id) && ProjectsUser.where(user_id: self.id, project_id: project_id).first.try(:admin?))
+  end
+
+  def project_member_permissions?(project_id, organization_id)
+    project_admin_permissions?(project_id, organization_id) || project_member?(project_id)
+  end
+
+  def organization_member?(organization_id)
+    OrganizationsUser.where(user_id: self.id, organization_id: organization_id).first.present?
+  end
+
+  def project_member?(project_id)
+    organization_id = Project.find(project_id).try(:organization_id)
+    organization_member?(organization_id) && ProjectsUser.where(user_id: self.id, project_id: project_id).first.present?
+  end
```

Comment:
I think that it's good idea store all above methods in some module, smth like UserRoles, because logically we are verifying User Roles. What do you think?